### DISCUSSION TITLE
Why zero hit distance in `PxVehiclePhysXRoadGeometryQueryUpdate` is interpreted as non hit?

### DISCUSSION THREAD

**Discussion by Tooflevor** [NVIDIA-Omniverse/PhysX#74] [Q&A] [answered]
`PxVehiclePhysXRoadGeometryQueryUpdate` in a file `physx\source\physxvehicle2\src\physxRoadGeometry\VhPhysXRoadGeometryFunctions.cpp` has specific check for zero hit distance in the raycast query
https://github.com/NVIDIA-Omniverse/PhysX/blob/4d33492253ab64c2246d80257ac4ce0f37b40b22/physx/source/physxvehicle2/src/physxRoadGeometry/VhPhysXRoadGeometryFunctions.cpp#L112-L131

Something similar is in the sweep query type:
https://github.com/NVIDIA-Omniverse/PhysX/blob/4d33492253ab64c2246d80257ac4ce0f37b40b22/physx/source/physxvehicle2/src/physxRoadGeometry/VhPhysXRoadGeometryFunctions.cpp#L156-L223
So, if hit distance is exactly zero, result is like there are no hit at all.

But according to `PxLocationHit` and documentation, zero hit distance is a valid query result. So, why is it ignored?

**Comment by msauter-nvidia** [comment, marked as answer]
In the sweep case, this looks like a bug and we will have a look at it. For raycasts, it's a different story. The ray does not start at the wheel center but at the top of the wheel (see PxVehicleComputeSuspensionRaycast). The motivation for this is to handle scenarios where the wheel at max compression penetrates into the ground. Penetrations are only handled up to a certain distance (the wheel diameter, approximately). Beyond that, it's treated as no collision which was just a design decision. Since distance=0 means initial overlap, we ignore it at that point (see PxLocationHit::hadInitialOverlap()) (but just to be clear, for raycasts it would mean the wheel is sort of penetrating the ground by a distance of a wheel diameter at this point).

**Reply by Tooflevor** [reply]
Yeah, for raycasts it seems okay.
Initially, I stumbled upon zero hit distance in sweep case, and I am more interested in this one. Raycast just gave me the thought that I don't understand something.
Do I need to make an issue or something?

**Reply by msauter-nvidia** [reply]
It's tracked as a bug on our side and will probably get tackled in 5.2. Thanks.
